### PR TITLE
Add Improved Closedfaced Helmets Patches

### PR DESCRIPTION
**Pull request by sean-kang**
This is to add Improved Closedfaced Helmets Patches.

mod page: https://www.nexusmods.com/skyrimspecialedition/mods/12405

### CODE REVIEW

File: masterlist.yaml [hunk 1]
```diff
@@ -7084,12 +7099,43 @@ plugins:
       # version: 1.58
       - crc: 0xD7423A4E
         util: 'SSEEdit v4.0.0'
+  - name: 'imp_helm_Immersive_Armors.esp'
+    url:
+      - link: 'https://www.nexusmods.com/skyrimspecialedition/mods/12405/'
+        name: 'Improved Closedfaced Helmets Patches'
+    req:
+      - 'imp_helm_legend.esp'
```
Comment (MacSplody) [inline masterlist.yaml]: LOOT checks a plugins header for its masters and will sort a plugin after its master automatically.
So `afters` and `req` aren't required for any modules listed as a master in a plugin header.
You can view a plugins masters in Mod Organizer, Wrye Bash or SSEEdit.

image

File: masterlist.yaml [hunk 2]
```diff
@@ -7084,12 +7099,43 @@ plugins:
...
+        util: 'SSEEdit v4.0.2'
   - name: 'imp_helm_IDPM.esp'
     url:
       - link: 'https://www.nexusmods.com/skyrimspecialedition/mods/12405/'
         name: 'Improved Closedfaced Helmets Patches'
+    req:
+      - 'imp_helm_legend.esp'
     after:
       - 'imp_helm_USSEP.esp'
+      - 'Improved Dragon Priest Masks.esp'
```
Comment (MacSplody) [inline masterlist.yaml]: Neither rule is required as `imp_helm_legend.esp` and `Improved Dragon Priest Masks.esp` are masters of `imp_helm_IDPM.esp`.

image

File: masterlist.yaml [hunk 3]
```diff
@@ -7084,12 +7099,43 @@ plugins:
...
       - crc: 0xD7423A4E
         util: 'SSEEdit v4.0.0'
+  - name: 'imp_helm_Immersive_Armors.esp'
+    url:
+      - link: 'https://www.nexusmods.com/skyrimspecialedition/mods/12405/'
+        name: 'Improved Closedfaced Helmets Patches'
+    req:
+      - 'imp_helm_legend.esp'
+    after:
+      - 'Hothtrooper44_ArmorCompilation.esp'
```
Comment (MacSplody) [inline masterlist.yaml]: Not required, `Hothtrooper44_ArmorCompilation.esp` is a master of this patch.

image

File: masterlist.yaml [hunk 4]
```diff
@@ -7084,12 +7099,43 @@ plugins:
...
+      - crc: 0x135E09DE
+        util: 'SSEEdit v4.0.2'
+  - name: 'imp_helm_USSEP.esp'
+    url:
+      - link: 'https://www.nexusmods.com/skyrimspecialedition/mods/12405/'
+        name: 'Improved Closedfaced Helmets Patches'
+    req:
+      - 'imp_helm_legend.esp'
+    after:
+      - 'Unofficial Skyrim Special Edition Patch.esp'
```
Comment (MacSplody) [inline masterlist.yaml:7120]: The requires for `imp_helm_legend.esp` is redundant, but the `after` for `Unofficial Skyrim Special Edition Patch.esp` could be replaced with a `req`. As USSEP is not listed as a master in the plugins header.

image

File: masterlist.yaml [hunk 5]
```diff
@@ -7084,12 +7099,43 @@ plugins:
...
       - 'imp_helm_USSEP.esp'
+      - 'Improved Dragon Priest Masks.esp'
+    clean:
+      # version: 1
+      - crc: 0x135E09DE
+        util: 'SSEEdit v4.0.2'
+  - name: 'imp_helm_USSEP.esp'
+    url:
+      - link: 'https://www.nexusmods.com/skyrimspecialedition/mods/12405/'
+        name: 'Improved Closedfaced Helmets Patches'
```
Comment (MacSplody) [inline masterlist.yaml]: You could combine the _url_ for all 3 patches using a regex.
The Lore Weapon Expansion Addons uses a regex if you need an example.
You'd still need to keep a seperate entry for the other metadata.

**Review by MacSplody — CHANGES_REQUESTED** [review]

File: masterlist.yaml [hunk 6]
```diff
@@ -7072,6 +7072,21 @@ plugins:
...
+      - <<: *patch3rdParty
+        subs:
+          - 'Improved Dragon Priest Masks SE'
+          - '[Improved Closedfaced Helmets Patches](https://www.nexusmods.com/skyrimspecialedition/mods/12405)'
+        condition: 'active("Improved Dragon Priest Masks.esp") and not active("imp_helm_IDPM.esp")'
+      - <<: *patch3rdParty
+        subs:
+          - 'Unofficial Skyrim Special Edition Patch'
+          - '[Improved Closedfaced Helmets Patches](https://www.nexusmods.com/skyrimspecialedition/mods/12405)'
+        condition: 'active("Unofficial Skyrim Special Edition Patch.esp") and not active("imp_helm_USSEP.esp")'
```
Comment (MacSplody) [inline masterlist.yaml:7089]: Patch messages look perfect.

File: masterlist.yaml (hunk 1 above)
Comment (sean-kang) [reply masterlist.yaml]: Removed all the redundant dependencies.

**Comment by sean-kang** [comment]
All requested changes have been implemented. Thanks for the comments.

**Review by MacSplody — APPROVED** [review]

**Comment by MacSplody** [comment]
👍 Thanks.

**Pull request merged**